pycqed/instrument_drivers/physical_instruments/ZurichInstruments/ZI_HDAWG8.py

```diff
@@ -1,1128 +1,460 @@
-            self._snapshot_whitelist.update({'awgs_{}_enable'.format(i), 
-                'awgs_{}_outputs_0_amplitude'.format(i), 
-                'awgs_{}_outputs_1_amplitude'.format(i), 
-                'awgs_{}_sequencer_program_crc32_hash'.format(i)})
-
-        for i in range(8): 
-            self._snapshot_whitelist.update({
-                'sigouts_{}_direct'.format(i), 'sigouts_{}_offset'.format(i),
-                'sigouts_{}_on'.format(i) , 'sigouts_{}_range'.format(i)})
-
-
-        self._params_to_exclude = set(self.parameters.keys()) - self._snapshot_whitelist
-
-        self.connect_message(begin_time=t0)
-
-    def _add_extra_parameters(self):
-        self.add_parameter('timeout', unit='s',
-                           initial_value=10,
-                           parameter_class=ManualParameter)
-        self.add_parameter(
-            'cfg_num_codewords', label='Number of used codewords', docstring=(
-                'This parameter is used to determine how many codewords to '
-                'upload in "self.upload_codeword_program".'),
-            initial_value=self._num_codewords,
-            # N.B. I have commentd out numbers larger than self._num_codewords
-            # see also issue #358
-            vals=vals.Enum(2, 4, 8, 16, 32),  # , 64, 128, 256, 1024),
-            parameter_class=ManualParameter)
-
-        self.add_parameter(
-            'cfg_codeword_protocol', initial_value='identical',
-            vals=vals.Enum('identical', 'microwave', 'flux'), docstring=(
-                'Used in the configure codeword method to determine what DIO'
-                ' pins are used in for which AWG numbers.'),
-            parameter_class=ManualParameter)
-
-        for i in range(4):
-            self.add_parameter(
-                'awgs_{}_sequencer_program_crc32_hash'.format(i),
-                parameter_class=ManualParameter,
-                initial_value=0, vals=vals.Ints())
-
-
-    def snapshot_base(self, update: bool=False,
-                      params_to_skip_update =None, 
-                      params_to_exclude = None ):
-        """
-        State of the instrument as a JSON-compatible dict.
-        Args:
-            update: If True, update the state by querying the
-                instrument. If False, just use the latest values in memory.
-            params_to_skip_update: List of parameter names that will be skipped
-                in update even if update is True. This is useful if you have
-                parameters that are slow to update but can be updated in a
-                different way (as in the qdac)
-        Returns:
-            dict: base snapshot
-        """
-
-
-        if params_to_exclude is None: 
-            params_to_exclude = self._params_to_exclude
-
-        snap = {
-            "functions": {name: func.snapshot(update=update)
-                          for name, func in self.functions.items()},
-            "submodules": {name: subm.snapshot(update=update)
```

Comment:
We have added a part to the snaphsot here to make sure not all parameters are stored in our datafailes.

Could you add the  redefined snapshot_base to the ZI base class and the `self._params_to_exclude`  to this driver?

Reply:
Done.